src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
I don't know if I like the "everything is a helper" idea. Why should we have `exists` if we have `existsWithPromise`? What if `data` is only sometimes a promise? How are developers going to remember all of these helpers? It seems that this solution is good for the project maintainers at the expense of the project users.